src/transformers/models/whisper/generation_whisper.py

```diff
@@ -0,0 +1,1436 @@
+
+        >>> inputs = processor(ds[0]["audio"]["array"], return_tensors="pt")
+        >>> input_features = inputs.input_features
+
+        >>> generated_ids = model.generate(inputs=input_features)
+
+        >>> transcription = processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
+        >>> transcription
+        ' Mr. Quilter is the apostle of the middle classes, and we are glad to welcome his gospel.'
+        ```
+
+        """
+        # 0. deprecate old inputs
+        if "inputs" in kwargs:
+            input_features = kwargs.pop("inputs")
+            warnings.warn(
+                "The input name `inputs` is deprecated. Please make sure to use `input_features` instead.",
+                FutureWarning,
+            )
+        # 1. copy generation config
+        if generation_config is None:
+            generation_config = copy.deepcopy(self.generation_config)
+        else:
+            generation_config = copy.deepcopy(generation_config)
+
+        # 2. set global generate variables
+        input_stride = self.model.encoder.conv1.stride[0] * self.model.encoder.conv2.stride[0]
+        num_segment_frames = input_stride * self.config.max_source_positions
+        total_input_frames = self._retrieve_total_input_frames(
+            input_features=input_features, input_stride=input_stride, kwargs=kwargs
+        )
+        is_shortform = total_input_frames <= num_segment_frames
+
+        # 3. Make sure generation config is correctly set
+        # Make sure the generation config is correctly set depending on whether timestamps are to be returned or not
+        self._set_return_outputs(
+            return_dict_in_generate=return_dict_in_generate,
+            return_token_timestamps=return_token_timestamps,
+            is_shortform=is_shortform,
+            logprob_threshold=logprob_threshold,
+            generation_config=generation_config,
+        )
+        self._set_return_timestamps(
+            return_timestamps=return_timestamps, is_shortform=is_shortform, generation_config=generation_config
+        )
+        self._set_language_and_task(
+            language=language, task=task, is_multilingual=is_multilingual, generation_config=generation_config
+        )
+        # pass self.config for backward compatibility
+        self._set_forced_decoder_ids(
+            task=task,
+            language=language,
+            prompt_ids=prompt_ids,
+            generation_config=generation_config,
+            config=self.config,
+            kwargs=kwargs,
+        )
+        self._set_token_ids(generation_config=generation_config, config=self.config, kwargs=kwargs)
+        self._set_num_frames(
+            return_token_timestamps=return_token_timestamps, generation_config=generation_config, kwargs=kwargs
+        )
+
+        # 4. Retrieve logits processors
+        # => TODO(Patrick): The way `num_start_tokens` is retrieved here is too brittle. Need a better approach
```

Comment:
```suggestion
        # TODO(Patrick): The way `num_start_tokens` is retrieved here is a bit brittle. Once forced decoder_ids are refactored this could be improved.
```